app/SettingsScreen.js

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react';
+import { View, Text, Button, Switch, StyleSheet, Alert, TextInput, AsyncStorage } from 'react-native';
+
+export default class SettingsScreen extends Component{
+	state = {
+		blackBackground: false,
+		username: '',
+	}
+
+	componentDidMount(){
+		this.storageToState();
+	}
+
+	storageToState = async () => { 
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_background');
+  			if (value !== null){
+    			this.setState({blackBackground: value == 'true'});
+  			}
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert(JSON.stringify(error));
+		}
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_username');
+  			if (value !== null){
+    			this.setState({username: value});
+  			}
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('get username error');
+		}
+	}
+
+	toggleSwitch = (value) => {
+		this.setState({blackBackground: value});
+		this.storeBackground(value);
+	}
+
+	storeBackground = async (value) => {
+		try {
+  			await AsyncStorage.setItem('@DemoRNStore:settings_background', value.toString());
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('put background error');
+		}
+	}
+
+	onUserNameChanged = (text) => {
+		this.setState({username: text});
+		this.storeUserName(text);
+	}
+
+	storeUserName = async (username) => {
+		try {
+  			await AsyncStorage.setItem('@DemoRNStore:settings_username', username);
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('put username error');
+		}
+	}
+
+	containerStyle = () => {
+		let container = {
+			flex: 1,
+		};
+		if (this.state.blackBackground) {
+			container.backgroundColor = 'black';
+		}
+		return container;
+	}
+
+	render(){
+		let { blackBackground } = this.state;
+		return(
+		<View style={ this.containerStyle() }>
```

Comment:
changing component style by assigning function to it is not good. Better would be in this case to write:
style={this.state.blackBackground ? styles.blackBackground : styles.defaultBackground} and add those styles in StyleSheet.create().